src/sbtc-roadmap.md

```diff
@@ -0,0 +1,108 @@
+
+## sBTC SDK Core
+
+### Core Stacks types
+This phase of development is dedicated to implementing the fundamental types that sBTC developers require to interact effectively with the Stacks blockchain. With these key primitives in place, developers should be fully equipped to construct arbitrary Stacks transactions, establish connections with Stacks nodes, broadcast their transactions, and monitor the status of these transactions on the blockchain.
+
+### sBTC Transactions
+In this next phase, our goal is to furnish a comprehensive toolkit for creating and decoding all sBTC-specific transactions on both the Bitcoin and Stacks blockchains. This toolkit is designed to streamline the process of constructing sBTC operations in either OP_RETURN or commit-reveal format to Bitcoin, as well as initiating contract calls to sBTC contract functions on the Stacks blockchain. This development will allow developers to engage with sBTC operations with ease and efficiency.
+
+## Signer SDK
+
+### Signer p2p protocol
+The Signer peer-to-peer (p2p) protocol lays out the rules governing how signers discover and interact with each other. This initiative aims to precisely define this protocol, enabling any developer to implement their own signers, seamlessly connect with others, and engage in fluid communication. Furthermore, we aim to integrate essential types and methods into the sBTC SDK, simplifying the development of signers capable of communicating over this protocol.
+
+### ROAST signing rounds
+ROAST, a wrapper around FROST, is our chosen method for the signature rounds. This task involves constructing primitives that allow developers to effortlessly initiate signing rounds with ROAST and aggregate signatures for arbitrary messages.
+
+### Signer transactions
+With the basic signer primitives for ROAST and p2p communication established, this phase involves building a toolkit to construct and sign the specific transactions required by signers within the [sBTC 0.1](sbtc-dev.md) protocol. This toolkit should facilitate developers in creating and initiating a signing round for an sBTC fulfillment transaction, as well as the signer handover transaction.
+
+### Signer chainstate
+Signers need to maintain an overview of pending sBTC operations and a chain state to validate incoming signature requests and determine if they need to initiate a signing round. This phase involves constructing an initial database schema for signers to manage, and providing functions to connect to Bitcoin and Stacks nodes, ensuring they can maintain this local state view.
+
+### Signer leader election
+Since all signers collectively observe incoming sBTC requests and need to respond, they also need to agree on who should initiate and aggregate signatures for a particular request. This task involves defining this aspect of the signer protocol and expanding the signer SDK with functions to execute leader election for specific requests.
```

Comment:
Yeah, I'm curious to see if we can reuse some logic in between. One noteworthy difference though, afaik Stacks only chooses a single leader per block with no fallback - since all leaders have already committed to their blocks. The signer leader may not initiate a signing round or submit their aggregate signature, so this leader election process must also define an ordering so that a secondary leader can take over if the primary fails etc.

Reply:
I added a paragraph to highlight this now.

Reply:
I think the _single-leader_ description is a little misleading because if the leader fails there is another node that jumps in in the current design. IMHO I will describe it only as single leader if the block is skipped or empty if leader fails due to network disconnection, but is not in the current spec.

Reply:
Okay, then my understanding of stacks leader election is wrong. I'll have to re-read SIP-001.

Reply:
So from SIP-001 we have
> The existence of multiple chain tips is a direct consequence of the **single-leader design of the Stacks blockchain**. Because anyone can become a leader, this means that even misbehaving leaders can be selected. If a leader crashes before it can propagate its block data, or if it produces an invalid block, then no block will be appended to the leader's selected chain tip during its epoch.

I'm pretty sure this is still how leader election works in Stacks, but please point me to any updated design doc where the fallback mechanism would be described if I'm missing anything.